### PR TITLE
 processing data in models

### PR DESCRIPTION
**Pull request by BouaikIlham**
## In this pull request:
 - [ ] Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
 - [ ] Set up associations between models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+end
```
Comment (ShoiraTa) [inline app/models/user.rb]: - [ ] Kindly add a method  in User model that returns 3 most recent posts. eg: 

```ruby
def recent_post
  posts.order('created_at Desc').limit(3)
end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: :User
+  has_many :comments
+  has_many :likes
+end
```
Comment (ShoiraTa) [inline app/models/post.rb]: - [ ]  Kindly add a method in Post model that that updates the posts counter for a user.
- [ ]  Kindly add a method in Post model that that returns the 5 most recent comments for a given post.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: :User
+  belongs_to :posts
+end
```
Comment (ShoiraTa) [inline app/models/like.rb]: - [ ] Kindly add a method in Like model  that updates that updates the likes counter for a post.

eg:

```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: :User
+  belongs_to :post
+end
```
Comment (ShoiraTa) [inline app/models/comment.rb]: - [ ] Kindly add a method in Comment model  that updates the comments counter for a post.

eg:
```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @BouaikIlham ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by aimemalaika — APPROVED** [review]
Hi @BouaikIlham , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉

- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**